probe-rs/src/bin/probe-rs/util/common_options.rs

```diff
@@ -569,7 +569,22 @@ impl From<std::io::Error> for OperationError {
 
 impl From<OperationError> for DebuggerError {
     fn from(e: OperationError) -> Self {
-        DebuggerError::Other(e.into())
+        match e {
+            OperationError::AttachingFailed {
+                source,
+                connect_under_reset,
+            } => match source {
+                probe_rs::Error::Timeout => {
+                    if !connect_under_reset {
+                        DebuggerError::UserMessage(format!("{source} This can happen if the target is not in a state where it can be attached to. A hard reset during attaching might help. If your probe supports this option, try setting `connectUnderReset=true` in your `launch.json`."))
```

Comment:
`connectUnderReset=true` is not valid JSON, and I think we should not assume that everybody is using vscode.